### DISCUSSION TITLE
Using Native Toast Notifications for Windows with Avalonia 11

### DISCUSSION THREAD

**Discussion by sessiontalk** [AvaloniaUI/Avalonia#12502] [General]
Hi 

I have the following code in my Avalonia 11 app

  var builder = new ToastContentBuilder()
                   .AddArgument("conversationId", 9813)
                   .AddText("Some text")
                   .AddButton(new ToastButton()
                       .SetContent("Archive")
                       .AddArgument("action", "archive"));
                builder.Show();

The Show method is not  recognised after including  the package Microsoft.Toolkit.Uwp.Notifications;

Looking through various forums online, the Target framework should be set so I've tried net7.0-windows but that does
not resolve it. We are using net7 for our apps so the DesktopNotifications sample  for Avalonia which uses net6 doesn't help.
I've tried moving that to net7 but then get the same error as above. 

There is also the Microsoft.Windows.AppNotifications.Builder package but not sure how to use that with Avalonia

Any advice would be most appreciated. Thank you

**Comment by cricketthomas** [comment]
Have you tried setting your target framework to `net8.0-windows10.0.19041.0` or `net7.0-windows10.0.19041.0` or another windows version?

**Comment by kyuranger** [comment]
I had met the same problem before in WPF that Windows 10 supports this while Windows 7 does not support this.

Finally, I found a strange solution by creating a console program only for Toast Notifications.

Then your program runs the console program in background.

All problems solved.
